File: shopping-parent/manager-web/src/main/resources/static/js/controller/brandController.js
app.controller("brandController",function ($scope,$controller,brandservice) {
    $controller('baseController',{$scope:$scope});//继承
    $scope.findAll=function () {
        brandservice.findAll().success(function (data) {
            $scope.brandList=data;
        });
    };
    $scope.findPage=function (page,row) {
        brandservice.findPage(page,row).success(function (data) {
            $scope.brandList=data.rows;
            $scope.paginationConf.totalItems=data.total;
        })
    };
    $scope.findPageByCondition=function(){

        brandservice.findPageByCondition($scope.paginationConf.currentPage,$scope.paginationConf.itemsPerPage,$scope.entityCondition)
            .success(function (data) {
            $scope.brandList=data.rows;
            $scope.paginationConf.totalItems=data.total;
        })
    };
    $scope.save=function () {
        var servicesave=null;
        if($scope.brand.id!=null){
            servicesave=brandservice.update;
        }else {
            servicesave=brandservice.add;
        }
        servicesave($scope.brand)
            .success(function (data) {
                if(data.success){
                    $scope.brand={};
                    $scope.ids=[];
                    $scope.reload();
                }else {
                    $scope.brand={};
                    alert("失败原因："+data.message);

                }
            });
    };
    $scope.findOne=function (id) {
        brandservice.findOne(id)
            .success(function (data) {
                if(data.id>0){
                    $scope.brand={id: data.id, name: data.name, firstChar: data.firstChar};
                }else {
                    alert("Id:"+id+"不存在");
                }
            });
    };
    $scope.deleteOne=function (id) {
       brandservice.deleteOne(id)
            .success(function (data) {
                if(data.success){
                    $scope.reload();
                }else{
                    alert("失败原因："+data.message);
                }
            })
    };
    $scope.deletemany=function () {
        brandservice.deleteMany($scope.ids)
            .success(function (data) {
                if(data.success){
                    $scope.ids=[];
                    $scope.reload();

                }else{
                    $scope.ids=[];
                    alert("失败原因："+data.message);
                }
            })
    };
})

File: shopping-parent/manager-web/src/main/java/com/nchu/manager/controller/BrandController.java
package com.nchu.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;

import com.nchu.model.PageResult;
import com.nchu.model.Result;
import com.nchu.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import com.nchu.model.Brand;
import java.util.List;

@RestController
@RequestMapping("/brand")
public class BrandController {
    @Reference
    private BrandService brandService;
    @RequestMapping("/findAll")
    public List<Brand> findAll(){
        List<Brand> brandList=null;
        brandList=brandService.findAllBrand();
        return brandList;
    }
    @RequestMapping("/findPage")
    public PageResult<Brand> findPage(int page, int row){
        return brandService.findPage(page,row);
    }

    @RequestMapping("/add")
    public Result add(@RequestBody Brand brand){
        Result result=null;
        try {
            Integer res=brandService.add(brand);
            if(res==0){
                result=new Result(false,"新增失败");
            }else {
                result=new Result(true,"新增成功");
            }
        }catch (Exception e){
            result=new Result(false,"新增失败");
        }
        return  result;
    }

    @RequestMapping("/update")
    public Result update(@RequestBody Brand brand){
        Result result=null;
        try {
            Integer res=brandService.update(brand);
            if(res==0){
                result=new Result(false,"修改失败1");
            }else {
                result=new Result(true,"修改成功");
            }
        }catch (Exception e){
            result=new Result(false,"修改失败2");
        }
        return  result;
    }
    @RequestMapping("/findOne")
    public Brand findOne(Long id){
        Brand brand=null;
        try {
            brand = brandService.selectOne(id);
            if(brand==null){
                brand=new Brand();
                brand.setId(-1L);
                brand.setName("没有id:"+id+"的brand");
            }
        }catch (Exception e){
            brand=new Brand();
            brand.setId(-2L);
            brand.setName("失败2");
        }
        return  brand;
    }
    @RequestMapping("deleteOne")
    public Result deleteOne(Long id){
        Result result=null;
        try {
            Integer res=brandService.delete(id);
            if(res==0){
                result=new Result(false,"删除失败1");
            }else {
                result=new Result(true,"删除成功");
            }
        }catch (Exception e){
            result=new Result(false,"删除失败2");
        }
        return  result;
    }
    @RequestMapping("deleteMany")
    public Result deleteMany(Long[] ids){
        Result result=null;
        try {
            Integer res=brandService.delete(ids);
            if(res==ids.length){
                result=new Result(true,"删除成功");
            }else {
                result=new Result(false,"删除失败1");
            }
        }catch (Exception e){
            result=new Result(false,"删除失败2");
        }
        return  result;
    }
    @RequestMapping("/findPageByEntity")
    public PageResult<Brand> findPageByEntity(Integer page, Integer row, @RequestBody Brand entity){
        PageResult brandList=null;
        brandList=brandService.findPage(page, row, entity);
        return brandList;
    }
}
